addons/overheating/functions/fnc_calculateCooling.sqf

```diff
@@ -30,6 +30,12 @@ if (_totalTime > 1800) exitWith {0};
 //So Area = 210 * 1.1 * (mass / 7850) = mass * 0.029427 (for steel near that diameter)
 
 private _barrelSurface = _barrelMass * 0.029427;
+private _convectionRate = 25;
+
+// increased convection rate if weapon in/under water
+if (((getPosASL ACE_player) select 2) < -1.5) then {
```

Comment:
Is there no weapon memory point? That would include weapon on back and such?

Reply:
I'm actually going to have to revisit this, because cooling only effects currentWeapon once you start swimming currentWeapon is nil